src/libraries/System.DirectoryServices.Protocols/src/System/DirectoryServices/Protocols/ldap/LdapConnection.Linux.cs

```diff
@@ -24,17 +25,26 @@ private int InternalConnectToServer()
 
         private int InternalBind(NetworkCredential tempCredential, SEC_WINNT_AUTH_IDENTITY_EX cred, BindMethod method)
         {
-            int error;
             if (tempCredential == null && (AuthType == AuthType.External || AuthType == AuthType.Kerberos))
             {
-                error = BindSasl();
+                return BindSasl();
             }
-            else
+
+            if (tempCredential != null)
             {
-                error = Interop.ldap_simple_bind(_ldapHandle, cred.user, cred.password);
+                var tempDomainName = new StringBuilder(100);
```

Comment:
This might be a dangerous assumption, when AuthType is basic that may be fine, but here we may have cases were people try to use credentials in the form user@DOMAIN.COM which will basically be broken by this case, or people could have also already passed in a credential which already was in the form DOMAIN\\user which would now be broken as well as we would pass in DOMAIN\\DOMAIN\\user

Reply:
Except by choosing AuthType External or Kerberos you run into `ldap_simple_bind`, which always uses LDAP_AUTH_SIMPLE.
I figure out it by looking at [ldap_simple_bind_s](https://github.com/openldap/openldap/blob/master/contrib/ldaptcl/neoXldap.c#L515) and [ldap_bind](https://github.com/openldap/openldap/blob/master/libraries/libldap/bind.c#L66)

If people pass credentials in DOMAIN\user form and for some reason the project is moved to a Windows environment they will get Credential Error. It's whats happening now for those environments who isn't joined in the realm.

> or people could have also already passed in a credential which already was in the form DOMAIN\user which would now be broken as well as we would pass in DOMAIN\DOMAIN\user

Actually this same situation could happen in Windows, if developer choose Basic and user pass credential as DOMAIN\user. It's concating DOMAIN\DOMAIN\user. Maybe some specific validation whether user already typed domain?

This assumption is a try to minimize impact of diff's between Linux and Windows environments.